src/renderer.cpp

```diff
@@ -20,30 +20,54 @@
 
 #include "app.h"
 
-void gl_renderer::draw_spline(const std::vector<glm::vec3>& points, glm::mat4 vp, glm::vec3 colour) const {
+void gl_renderer::draw_spline(const std::vector<glm::vec3>& points, const glm::mat4& vp, const glm::vec3& colour) const {
 	
 	if(points.size() < 1) {
 		return;
 	}
 	
 	std::vector<float> vertex_data;
-	for(std::size_t i = 0; i < points.size() - 1; i++) {
-		vertex_data.push_back(points[i].x);
-		vertex_data.push_back(points[i].y);
-		vertex_data.push_back(points[i].z);
-		
-		vertex_data.push_back(points[i].x + 0.5);
-		vertex_data.push_back(points[i].y);
-		vertex_data.push_back(points[i].z);
-		
-		vertex_data.push_back(points[i + 1].x);
-		vertex_data.push_back(points[i + 1].y);
-		vertex_data.push_back(points[i + 1].z);
+
+	for(auto iter = points.begin(); iter != points.end()-1; iter++) {
+		vertex_data.push_back((*iter).x);
+		vertex_data.push_back((*iter).y);
+		vertex_data.push_back((*iter).z);
+		vertex_data.push_back((*(iter+1)).x);
+		vertex_data.push_back((*(iter+1)).y);
+		vertex_data.push_back((*(iter+1)).z);
 	}
-	draw_tris(vertex_data, vp, colour);
```

Comment:
Since you're using GL_LINE_STRIP, this whole loop should be unnecessary. You can just pass the contents of the points buffer to OpenGL directly. Also call glLineWidth somewhere so it isn't super hard to click on. Maybe only do it do the pickframe.

Reply:
Yes, I'll see about this. It might be better to use a thicker line, but that's just shoving the problem under the rug imo. The selection update will be better. Also, if you go around and select a few different objects, it seems to work for some of them. Not sure what that's all about.